entoas/edges_test.go

```diff
@@ -0,0 +1,88 @@
+package entoas
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"entgo.io/contrib/entoas/serialization"
+	"entgo.io/ent/entc"
+	"entgo.io/ent/entc/gen"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWalk_CycleDepth(t *testing.T) {
+	w := walk{}
+	require.Equal(t, uint(0), w.cycleDepth())
+
+	w = walk{"a"}
+	require.Equal(t, uint(1), w.cycleDepth())
+
+	w = walk{"a", "b"}
+	require.Equal(t, uint(1), w.cycleDepth())
+
+	w = walk{"a", "a"}
+	require.Equal(t, uint(2), w.cycleDepth())
+
+	w = walk{"a", "b", "b"}
+	require.Equal(t, uint(2), w.cycleDepth())
+
+	w = walk{"a", "b", "b", "c"}
+	require.Equal(t, uint(1), w.cycleDepth())
+
+	w = walk{"a", "b", "b", "a"}
+	require.Equal(t, uint(2), w.cycleDepth())
+
+	w = walk{"a", "a", "b", "a"}
+	require.Equal(t, uint(3), w.cycleDepth())
+}
+
+func TestWalk_Push(t *testing.T) {
+	w := walk{}
```

Comment:
ditto

Reply:
I don't know what you mean here, sorry :crying_cat_face: